app/src/main/java/org/wikipedia/talk/TalkTopicsActivity.kt

```diff
@@ -166,12 +170,21 @@ class TalkTopicsActivity : BaseActivity() {
         }
         notificationButtonView = NotificationButtonView(this)
         Prefs.hasAnonymousNotification = false
-        talkTopicsProvider = TalkTopicsProvider(pageTitle)
+
+        lifecycleScope.launchWhenCreated {
+            viewModel.uiState.collect {
+                when (it) {
+                    is TalkTopicsViewModel.UiState.Success -> updateOnSuccess(it.pageTitle, it.discussionToolsInfoResponse, it.lastModifiedResponse)
+                    is TalkTopicsViewModel.UiState.Error -> updateOnError(it.throwable)
+                }
+            }
+        }
+
+        viewModel.loadTopics(pageTitle)
```

Comment:
The ViewModel should call `loadTopics()` on its own when it's created. (We want to avoid calling `loadTopics()` every time the activity is recreated.)

Reply:
Yes! Makes sense to me.